src/zcash/address/orchard.hpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021 The Zcash Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_ADDRESS_ORCHARD_H
+#define ZCASH_ADDRESS_ORCHARD_H
+
+#include "streams.h"
+#include "rust/orchard/keys.h"
+
+namespace libzcash {
+
+class OrchardRawAddress
```

Comment:
Add unit tests that check the round-trip serialization of these types (which would have caught the FFI interface mismatch).